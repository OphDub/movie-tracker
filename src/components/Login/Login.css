.error {
  border: 1px solid red;
}

.Login {
  background: linear-gradient(#336E7B, #000000);
  min-height: 100vh;
  padding: 3em 0;
  margin: 0px;
  display: flex;
  flex-direction: column;
}

.login-prompt {
  align-self: center;
  padding: 0.5em;
  color: #ffffff;
}

.login-form {
  display: flex;
  flex-direction: column;
  align-self: center;
  width: 30%;
  padding-bottom: 2em;
}

.login-btn {
  border: none;
  border-style: none;
}

.login-btn:hover {
  background: #000000;
  color: #e3be42;
  border: none;
}