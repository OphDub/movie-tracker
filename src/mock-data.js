 export const cleanMovieArray = [
    {
      movie_id: 399055,
      title: 'The Shape of Water',
      vote_average: 7.4,
      vote_count: 794,
      overview: 'An other-worldly story, set against the backdrop of Cold War era America circa 1962, where a mute janitor working at a lab falls in love with an amphibious man being held captive there and devises a plan to help him escape.',
      release_date: '2017-12-08',
      poster_path: 'https://image.tmdb.org/t/p/w500//iLYLADGA5oKGM92Ns1j9CDgk3iI.jpg'
    },
    {
      "movie_id": 284054,
      "title": "Black Panther",
      "vote_average": 5,
      "vote_count": 72,
      "overview": "T'Challa, the new ruler of the advanced kingdom of Wakanda, must defend his land from being torn apart by enemies from outside and inside the country.",
      "release_date": "2018-02-13",
      "poster_path": "https://image.tmdb.org/t/p/w500//bLBUCtMQGJclH36clliPLmljMys.jpg"
    }
] 

export const dirtyMovieObject = { results: [
    { 
      vote_average: 7.4,
      overview: 'An other-worldly story, set against the backdrop of Cold War era America circa 1962, where a mute janitor working at a lab falls in love with an amphibious man being held captive there and devises a plan to help him escape.',
      vote_count: 794,
      title: 'The Shape of Water',
      poster_path: '/iLYLADGA5oKGM92Ns1j9CDgk3iI.jpg',
      release_date: '2017-12-08',
      id: 399055,
      randomnumber: 24341324142134,
    },
    {
      "title": "Black Panther",
      "vote_count": 72,
      "thing": "asdfasdfasdf",
      "id": 284054,
      "overview": "T'Challa, the new ruler of the advanced kingdom of Wakanda, must defend his land from being torn apart by enemies from outside and inside the country.",
      "release_date": "2018-02-13",
      "poster_path": "/bLBUCtMQGJclH36clliPLmljMys.jpg",
      "vote_average": 5,
    }
]}

export const oneMovie = {
  movie_id: 284054,
  title: 'Coco',
  vote_average: 7.7,
  vote_count: 1987,
  overview: 'Despite his familyâ€™s baffling generations-old ban on music, Miguel dreams of becoming an accomplished musician like his idol, Ernesto de la Cruz. Desperate to prove his talent, Miguel finds himself in the stunning and colorful Land of the Dead following a mysterious chain of events. Along the way, he meets charming trickster Hector, and together, they set off on an extraordinary journey to unlock the real story behind Miguel\'s family history.',
  posterImage: 'https://image.tmdb.org/t/p/w500//eKi8dIrr8voobbaGzDpe8w0PVbC.jpg'
}

export const userData = {
  id: 2,
  name: 'brophus park',
  email: 'bros@gmail.com',
  password: 'pass',
}

export const movieRawData = {
  data: {
      movie_id: 399055,
      title: 'The Shape of Water',
      vote_average: 7.4,
      vote_count: 794,
      overview: 'An other-worldly story, set against the backdrop of Cold War era America circa 1962, where a mute janitor working at a lab falls in love with an amphibious man being held captive there and devises a plan to help him escape.',
      release_date: '2017-12-08',
      poster_path: 'https://image.tmdb.org/t/p/w500//iLYLADGA5oKGM92Ns1j9CDgk3iI.jpg'
    }
}
