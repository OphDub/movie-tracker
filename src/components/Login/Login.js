import React, {Component} from 'react';
import { validateUser, getAllFavorites } from '../../api.js';
import { connect } from 'react-redux';
import { activeUserAction, addFavoriteAction } from '../../actions/actionIndex';
import SignUp from '../SignUp/SignUp';
import PropTypes from 'prop-types';
import './Login.css';

export class Login extends Component {
  constructor(props){
    super(props);
    this.state = {
      username: '',
      password: '',
      errorStatus: false
    }
  }

  handleChange=(e)=> {
    const { name, value } = e.target;
    this.setState({[name]:value});
  }

  handleLoginAttempt = async (e) => {
    e.preventDefault();
    const { username, password } = this.state;
    const validate = await validateUser(username, password);

    if (validate.status === 'success') {
      this.props.handleLogin(validate.data);
      this.getFavorites(validate.data)
      this.setState({username: '', password: ''});
    } else {
      console.log('failed to login try again n00b')
      this.setState({errorStatus: true})
    }
  }



  getFavorites = async (user) => {
    const allFavs = await getAllFavorites(user.id);
    this.props.addFavorite(allFavs.data);
  }

  render() {
    return (
      <div>
        <form type="submit">
          <input
            className={!this.state.errorStatus ? "" : "error"}
            name="username"
            type='text'
            placeholder="User Name or E-mail"
            value={this.state.username}
            onChange={this.handleChange}
            />
          <input
            className={!this.state.errorStatus ? "" : "error"}
            name="password"
            type='password'
            placeholder="Password"
            value={this.state.password}
            onChange={this.handleChange}
            />
          <button
            onClick={this.handleLoginAttempt}
            type="submit">login
          </button>
          {this.state.errorStatus && 
            <span className="login-error">E-mail or password do not match, try again</span>
          }
        </form>

        <SignUp />
      </div>
    )
  }
}

const mapStateToProps = (state) => ({
  activeUser: state.activeUser,
});

const mapDispatchToProps = (dispatch) => ({
  handleLogin: (user) => dispatch(activeUserAction(user)),
  addFavorite: (favoriteData) => dispatch(addFavoriteAction(favoriteData)),
});

export default connect(mapStateToProps, mapDispatchToProps)(Login);

Login.propTypes = {
  addFavorite: PropTypes.func.isRequired ,
  handleLogin: PropTypes.func.isRequired
}
