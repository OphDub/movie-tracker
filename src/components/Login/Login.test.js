/* eslint-disable */
import React from 'react';
import { Login } from './Login';
import { shallow } from 'enzyme';
import { userData, cleanMovieArray } from '../../mock-data.js';

describe('Login', () => {
  let wrapper;

  beforeEach( () => {
    wrapper = shallow(
      <Login
        activeUser={userData}
      />);
  });

  it('should match the snapshot test', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('handleChange should update state when it is called', () => {
    const mockEvent = { target: {name: 'name' ,value: 'brophus park'}};

    wrapper.instance().handleChange(mockEvent);
    wrapper.update();

    expect(wrapper.state().name).toEqual(mockEvent.target.value);
  });

  it('should call getAllFavorites from api and addFavorite when getFavorites is called', async () => {
    window.fetch = jest.fn().mockImplementation(() => {
      return new Promise((resolve, reject) => {
        resolve({
          response: {results: cleanMovieArray}
        });
      });
    });

    expect(wrapper.instance().getFavorites(userData)).resolves.toEqual(cleanMovieArray)
  });
});