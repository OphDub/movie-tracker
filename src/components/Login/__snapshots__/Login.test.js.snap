// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should match the snapshot test 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login
    activeUser={
      Object {
        "email": "bros@gmail.com",
        "id": 2,
        "name": "brophus park",
        "password": "pass",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <form
          type="submit"
        >
          <input
            className=""
            name="username"
            onChange={[Function]}
            placeholder="User Name or E-mail"
            type="text"
            value=""
          />
          <input
            className=""
            name="password"
            onChange={[Function]}
            placeholder="Password"
            type="password"
            value=""
          />
          <button
            onClick={[Function]}
            type="submit"
          >
            login
          </button>
        </form>,
        <Connect(SignUp) />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              className=""
              name="username"
              onChange={[Function]}
              placeholder="User Name or E-mail"
              type="text"
              value=""
            />,
            <input
              className=""
              name="password"
              onChange={[Function]}
              placeholder="Password"
              type="password"
              value=""
            />,
            <button
              onClick={[Function]}
              type="submit"
            >
              login
            </button>,
            false,
          ],
          "type": "submit",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "",
              "name": "username",
              "onChange": [Function],
              "placeholder": "User Name or E-mail",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "",
              "name": "password",
              "onChange": [Function],
              "placeholder": "Password",
              "type": "password",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "login",
              "onClick": [Function],
              "type": "submit",
            },
            "ref": null,
            "rendered": "login",
            "type": "button",
          },
          false,
        ],
        "type": "form",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <form
            type="submit"
          >
            <input
              className=""
              name="username"
              onChange={[Function]}
              placeholder="User Name or E-mail"
              type="text"
              value=""
            />
            <input
              className=""
              name="password"
              onChange={[Function]}
              placeholder="Password"
              type="password"
              value=""
            />
            <button
              onClick={[Function]}
              type="submit"
            >
              login
            </button>
          </form>,
          <Connect(SignUp) />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                className=""
                name="username"
                onChange={[Function]}
                placeholder="User Name or E-mail"
                type="text"
                value=""
              />,
              <input
                className=""
                name="password"
                onChange={[Function]}
                placeholder="Password"
                type="password"
                value=""
              />,
              <button
                onClick={[Function]}
                type="submit"
              >
                login
              </button>,
              false,
            ],
            "type": "submit",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "",
                "name": "username",
                "onChange": [Function],
                "placeholder": "User Name or E-mail",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "",
                "name": "password",
                "onChange": [Function],
                "placeholder": "Password",
                "type": "password",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "login",
                "onClick": [Function],
                "type": "submit",
              },
              "ref": null,
              "rendered": "login",
              "type": "button",
            },
            false,
          ],
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
